iModelCore/ECDb/ECDb/ECSql/PragmaECSqlPreparedStatement.cpp

```diff
@@ -0,0 +1,1092 @@
+            return BE_SQLITE_ERROR;
+        }
+        auto mapInfo = ecdb.Schemas().GetClassMapStrategy(cls.GetSchema().GetName(), cls.GetName());
+        if (mapInfo.IsNotMapped() || mapInfo.IsEmpty()) {
+            ecdb.GetImpl().Issues().ReportV(
+                IssueSeverity::Error,
+                IssueCategory::BusinessProperties,
+                IssueType::ECSQL, "PRAGMA %s, is not invoked on class %s which is not mapped or exist.", GetName().c_str(), cls.GetFullName());
+            return BE_SQLITE_ERROR;
+        }
+        if (val.GetBool()) {
+            m_disqualifiedClassSet.insert(cls.GetId());
+        } else {
+            m_disqualifiedClassSet.erase(cls.GetId());
+        }
+        rowSet = std::make_unique<StaticPragmaResult>(ecdb);
+        rowSet->FreezeSchemaChanges();
+        return BE_SQLITE_OK;
+    }
+    static std::unique_ptr<PragmaManager::Handler> Create () { return std::make_unique<DisqualifyTypeIndex>(); }
+};
+
+//================================================================================
+// @bsiclass PragmaECDbVersion
+//================================================================================
+struct PragmaECDbVersion : PragmaManager::GlobalHandler {
+    PragmaECDbVersion():GlobalHandler("ecdb_ver","return current and file profile versions"){}
+    ~PragmaECDbVersion(){}
+    virtual DbResult Read(PragmaManager::RowSet& rowSet, ECDbCR ecdb, PragmaVal const&)  override {
+        auto result = std::make_unique<StaticPragmaResult>(ecdb);
+        result->AppendProperty("current", PRIMITIVETYPE_String);
+        result->AppendProperty("file", PRIMITIVETYPE_String);
+        result->FreezeSchemaChanges();
+        auto row = result->AppendRow();
+        row.appendValue() = ECDb::CurrentECDbProfileVersion().ToString();
+        row.appendValue() =  ecdb.GetECDbProfileVersion().ToString();
+        rowSet = std::move(result);
+        return BE_SQLITE_OK;
+    }
+    virtual DbResult Write(PragmaManager::RowSet& rowSet, ECDbCR ecdb, PragmaVal const&) override {
+        ecdb.GetImpl().Issues().ReportV(IssueSeverity::Error, IssueCategory::BusinessProperties, IssueType::ECSQL, "PRAGMA %s is readonly.", GetName().c_str());
+        rowSet = std::make_unique<StaticPragmaResult>(ecdb);
+        rowSet->FreezeSchemaChanges();
+        return BE_SQLITE_READONLY;
+    }
+    static std::unique_ptr<PragmaManager::Handler> Create () { return std::make_unique<PragmaECDbVersion>(); }
+};
+
+//================================================================================
+// @bsiclass PragmaFileInfo
+//================================================================================
+struct PragmaFileInfo : PragmaManager::GlobalHandler {
+    PragmaFileInfo():GlobalHandler("file_info","return file info"){}
+    ~PragmaFileInfo(){}
+    void AppendDbGuid(ECDbCR ecdb, BeJsValue row) {
+        row.appendValue() = "db_guid";
```

Comment:
From an iModel point of view this should be iModel Id but at the ECDb level just calling it the db guid is correct.  

This highlights that this ECSql pragma blurs the line between ECDb and iModel concepts.  Maybe it would be better to focus on ECDb file information here, I think that has the most value anyway.

For example:
- The bis schema version is a nice to have but you can already get that via ECDbMeta queries.  
- Type is an iModel interpretation of the briefcase id, any one who knows what 'standalone' and 'briefcase' mean should also be able to interpret the id to figure that out.


I think at least schema version and type should be removed.

Reply:
removed pragma for file_info for now until we figure out what useful information, we can show in it.